File: app/src/main/java/com/kite/mnemoai/data/network/deepseek/model/DeepseekRequestBody.java
package com.kite.mnemoai.data.network.deepseek.model;

import com.kite.mnemoai.data.network.PromptConstants;

import java.util.ArrayList;
import java.util.List;

public class DeepseekRequestBody {

    private String model;
    private List<Message> messages = new ArrayList<>();

    private Thinking thinking;
    private ResponseFormat responseFormat = new ResponseFormat();

    private static Message systemMessage = new Message(PromptConstants.SYS_CONTENT, PromptConstants.SYS_ROLE);

    public DeepseekRequestBody(String word, boolean isEnableThinking) {
        this.messages.add(systemMessage);
        this.messages.add(new Message(word, PromptConstants.USER_ROLE));
        this.model = PromptConstants.DEEPSEEK_FLASH_MODEL;
        this.thinking = new Thinking(isEnableThinking);
    }

    static class Thinking{
        private String type;

        public Thinking(boolean enable) {
            this.type = enable ? "enabled" : "disabled";
        }
    }

    static class ResponseFormat {

        private String type;

        public ResponseFormat() {
            this.type = "json_object";
        }
    }
}

File: core/network/src/main/java/com/kite/mnemoai/network/deepseek/model/DeepseekRequestBody.kt
package com.kite.mnemoai.network.deepseek.model

import com.google.gson.annotations.SerializedName

data class DeepseekRequestBody(
    val messages: MutableList<Message>,
    val model: String,
    @field:SerializedName("thinking")
    val thinking: Thinking?,
    @field:SerializedName("reasoning_effort")
    val reasoningEffort: ReasoningEffort?,
    @field:SerializedName("response_format")
    val responseFormat: ResponseFormatConfig?,
)
